Framework/API/inc/MantidAPI/ISpectrum.h

```diff
@@ -54,6 +59,9 @@ class DLLExport ISpectrum {
 
   void copyInfoFrom(const ISpectrum &other);
 
+  /// Copy data from another ISpectrum with double-dynamic dispatch.
+  virtual void copyDataFrom(const ISpectrum &other) = 0;
```

Comment:
Might make things slightly clearer to name arguments `source` when function is `copyDataFrom` and `sink` when function is `copyDataInto`